storage_policies/stack/__main__.py

```diff
@@ -186,3 +186,77 @@ def add_bucket_permissions(name: str,
                            release_access_group,
                            release_bucket,
                            'roles/storage.objectViewer')
+
+# Cloud Run is used for the server component of the analysis-runner.
+cloudrun = gcp.projects.Service('cloudrun-service',
+                                service='run.googleapis.com',
+                                disable_on_destroy=False)
+
+analysis_runner_service_account = gcp.serviceaccount.Account(
+    'analysis-runner-service-account',
+    account_id='analysis-runner-server',
+    display_name='analysis-runner service account',
+    opts=pulumi.resource.ResourceOptions(depends_on=[cloudidentity]))
+
+secretmanager = gcp.projects.Service('secretmanager-service',
+                                     service='secretmanager.googleapis.com',
+                                     disable_on_destroy=False)
+
+# Add a secret that will hold the Hail token.
+hail_token_secret = gcp.secretmanager.Secret(
+    'hail-token-secret',
+    replication=gcp.secretmanager.SecretReplicationArgs(
+        user_managed=gcp.secretmanager.SecretReplicationUserManagedArgs(
```

Comment:
Just wondering, what's the reasoning to use the user-managed replication policy here?

Reply:
When you create a secret in the cloud console, the explanation is:

> To choose specific regions for storing your secret, select Manually select regions. If you do not select this, Google Cloud will choose the best regions for you. This setting cannot be changed after a secret is created.

I'm not sure how GCP chooses the "best" region. This just makes the selection explicit, so the data is colocated to where we run our workloads. In comparison to buckets, this probably doesn't matter much, but shouldn't hurt either :).

Reply:
Thanks for the explanation!